### PR TITLE
#5379 [Manager] Prevent users from refreshing the browser directly be…

### PR DESCRIPTION
**Pull request by eechih**
…fore the long-term task ends

### Checklist

- [x] [Review Contribution Guidelines](https://oharastream.github.io/en/docs/master/contributing/)
- [x] Add at least one reviewer (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add an assignee (of course, you are the assignee by default)
- [x] Do not set milestone, projects and labels (PR labels are added automatically by Jenkins)
- [x] Link to an issue #5379 
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (If the failed tests are unrelated to this PR, add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/cypress/it/appBar/appBar.test.ts [hunk 1]
```diff
@@ -176,6 +176,13 @@ describe('App Bar', () => {
       const workspaceName = 'workspace2';
       cy.visit('/');
 
+      // close the intro dialog
+      cy.get('body').then(($body) => {
+        if ($body.find('[data-testid="intro-dialog"]').length > 0) {
+          cy.findByTestId('close-intro-button').filter(':visible').click();
+        }
+      });
+
```
Comment (saivirtue) [inline ohara-manager/client/cypress/it/appBar/appBar.test.ts:185]: This test is under context "Add workspaces", i.e., we should have workspaces added already. The assertion of "Intro Dialog" may not need :)

File: ohara-manager/client/cypress/it/appBar/appBar.test.ts [hunk 2]
```diff
@@ -176,6 +176,13 @@ describe('App Bar', () => {
       const workspaceName = 'workspace2';
       cy.visit('/');
```
Comment (saivirtue) [inline ohara-manager/client/cypress/it/appBar/appBar.test.ts:177]: We add this command in "beforeEach" phase, this line could be removed.

File: ohara-manager/client/src/components/common/FSMStepper/index.js [hunk 3]
```diff
@@ -116,12 +119,30 @@ const FSMStepper = React.forwardRef((props, ref) => {
...
+        testId="abort-task-confirm-dialog"
+        title="Abort task?"
+      />
+      {!isFinish && (
+        <Beforeunload
+          onBeforeunload={() =>
+            'The task currently being executed is not yet complete. If aborted, this workspace will be UNAVAILABLE. Are you sure you want to abort?'
+          }
+        />
+      )}
```
Comment (saivirtue) [inline ohara-manager/client/src/components/common/FSMStepper/index.js:145]: We should add a test for this part logic

File: ohara-manager/client/cypress/it/appBar/appBar.test.ts (hunk 1 above)
Comment (eechih) [reply ohara-manager/client/cypress/it/appBar/appBar.test.ts:185]: it.only() 單獨執行這個測試，會跳出 intro-dialog，只是為了測試方便 :)

File: ohara-manager/client/cypress/it/appBar/appBar.test.ts (hunk 2 above)
Comment (eechih) [reply ohara-manager/client/cypress/it/appBar/appBar.test.ts:177]: will do

File: ohara-manager/client/src/components/common/FSMStepper/index.js (hunk 3 above)
Comment (eechih) [reply ohara-manager/client/src/components/common/FSMStepper/index.js:145]: will do

File: ohara-manager/client/cypress/it/appBar/appBar.test.ts (hunk 1 above)
Comment (saivirtue) [reply ohara-manager/client/cypress/it/appBar/appBar.test.ts:185]: Oh...I see.
Maybe you can convert this PR to "draft" so I won't expect this PR is "ready" 😅

Comment (eechih) [reply ohara-manager/client/cypress/it/appBar/appBar.test.ts:185]: I will move into `cy.createNode` command

File: ohara-manager/client/src/components/common/FSMStepper/index.js (hunk 3 above)
Comment (eechih) [reply ohara-manager/client/src/components/common/FSMStepper/index.js]: wow~ 目前不知道如何用 Cypress 去測試 window beforeunload event，暫時先不處理，我開了一個 Issue 來追蹤這項工作 #5381

**Comment by eechih** [comment]
retry manager-e2e

**Comment by eechih** [comment]
retry manager-it

**Pull request merged**